### ISSUE TITLE
could you please support api url?

### ISSUE THREAD

**Issue by lisasuosuo** [clairefro/obsidian-chat-cbt-plugin#13]
can I use other api transfel url and api to connect to the model?

**Comment by clairefro** [comment]
Do you have an API in mind? Every API has a different interface

**Comment by clairefro** [comment]
By the way, if using Ollama, you can use any model found on [Hugging Face](https://huggingface.co/models) locally. after downloading and running the model in Ollama, you can update the ChatCBT settings to point to that model. Hope this helps

**Comment by lisasuosuo** [comment]
Thanks for your reply very much !!! Actually the API transfer url I mean is like this.
```from openai import OpenAI

client = OpenAI(
    # defaults to os.environ.get("OPENAI_API_KEY")
    api_key="YOUR API KEY",
    base_url="https://api.chatanywhere.tech/v1"
    # base_url="https://api.chatanywhere.cn/v1"
)
```
It is the api transfer host. Because the region where I am can't use the openai api directly, we should use the host to transfer. Hopefully I've made it clear.

**Comment by clairefro** [comment]
@lisasuosuo oh! Thanks for the example. I think I can make this work.

In the meantime, if you'd like to try you can use Ollama with Mistral

**Comment by lisasuosuo** [comment]
Yeah , Thanks again for your quickly and helpful reply!!!  I will try your suggestion! Expected to use this plugin!

**Comment by Melodeiro** [comment]
There are a lot of services with openai-compatible API layer. For example, deepseek. Or you can set up litellm-server which will convert all known APIs to openai API. So setting the custom api url is a very convenient and potent option. Even better if there would be an option to configure a list of models, with an option to set url, token and model name for each of them, and option to switch them in the side-toolbar context menu to find out which model gives the better answers for a specific question.

But even with just a custom api url for openai api you can leverage cloud solutions, like the mentioned huggingface, because its inference API is openai-compatible. There is actually a lot more of cloud providers like that one with free quota as well

**Comment by clairefro** [comment]
Thanks for the feedback! I'm open to adding to adding broader model support, but I don't have the bandwidth at the moment. 

If you need immediate use of other models, try forking this repo and using @Melodeiro's solution of litellm-server (and feel free to make a PR if you get it working!)

**Comment by Melodeiro** [comment]
A better (and i believe easier to implement in case of supporting a broad model range) solution emerged recently: https://github.com/pfrankov/obsidian-ai-providers

**Comment by clairefro** [comment]
@Melodeiro thank you! I'm looking into this
